src/ostree/ot-builtin-static-delta.c

```diff
@@ -377,6 +409,84 @@ ot_static_delta_builtin_apply_offline (int argc, char **argv, OstreeCommandInvoc
   return TRUE;
 }
 
+static gboolean
+ot_static_delta_builtin_verify (int argc, char **argv, OstreeCommandInvocation *invocation, GCancellable *cancellable, GError **error)
+{
+  g_autoptr(GOptionContext) context = g_option_context_new ("STATIC-DELTA-FILE [KEY-ID...]");
+  g_autoptr(OstreeRepo) repo = NULL;
+  gboolean verified;
+  char **key_ids;
+  int n_key_ids;
+
+  if (!ostree_option_context_parse (context, verify_options, &argc, &argv, invocation, &repo, cancellable, error))
+    return FALSE;
+
+  if (argc < 3)
+    {
+      g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "DELTA must be specified");
+      return FALSE;
+    }
+
+  if (opt_sign_name == NULL)
+    opt_sign_name = "ed25519";
+
+  const char *delta_id = argv[2];
+
+  if (!ostree_repo_static_delta_is_signed (repo, delta_id, error))
+    {
+      if (*error)
+        {
+          g_debug ("Unable to find static-delta signature(s): %s", (*error)->message);
+          g_clear_error (error);
+        }
+      g_print ("Not signed\n");
+      return TRUE;
+    }
+
+  g_autoptr(OstreeSign) sign = ostree_sign_get_by_name (opt_sign_name, error);
+  if (!sign)
+    {
+      g_print("Sign-type not supported\n");
+      return FALSE;
+    }
+
+  key_ids = argv + 2;
+  n_key_ids = argc - 2;
+  for (int i = 0; i < n_key_ids; i++)
+    {
+      g_autoptr (GVariant) pk = g_variant_new_string(key_ids[i]);
+      ostree_sign_add_pk(sign, pk, error);
+    }
+  if ((n_key_ids == 0) || opt_filename)
+    {
+      g_autoptr (GVariantBuilder) builder = NULL;
+      g_autoptr (GVariant) options = NULL;
+
+      builder = g_variant_builder_new (G_VARIANT_TYPE ("a{sv}"));
+      /* Use custom directory with public and revoked keys instead of system-wide directories */
+      if (opt_keysdir)
+        g_variant_builder_add (builder, "{sv}", "basedir", g_variant_new_string (opt_keysdir));
+      /* The last chance for verification source -- system files */
+      if (opt_filename)
+        g_variant_builder_add (builder, "{sv}", "filename", g_variant_new_string (opt_filename));
+      options = g_variant_builder_end (builder);
+
+      if (!ostree_sign_load_pk (sign, options, error))
+        return FALSE;
+    }
+
+  verified = ostree_repo_static_delta_verify_signature (repo, delta_id, sign, error);
```

Comment:
`return ostree_repo_static_delta_verify_signature()` -- otherwise it would return success on failed verification and problems with usage in shell scripts/other progs.